### DISCUSSION TITLE
Make comments

### DISCUSSION THREAD

**Discussion by jgVitc** [stephanrauh/ngx-extended-pdf-viewer#1375] [Q&A]
Hi,
Is there a way to all the user add comments (even multiple) to the PDF?

**Comment by stephanrauh** [comment]
Not yet. It's beyond my scope (not to mention my limited available leisure time - sigh!). However, if you're willing to implement the feature yourself and to send me a pull-request or a recipe how to implement it, I'll happily support you.

Basically, all you have to do is to add the user's comments to the array of annotations. That's probably challenging enough, but with some effort and time, it's not that difficult. However, what I don't know is how to add these annotations to the PDF file. I suppose you want to be able to save the PDF file including the annotations. I'm not familiar with that part of the implementation of pdf.js, so I expect that to be the difficult part.

**Reply by jgVitc** [reply]
Thank you.  I will contact you if I have questions and decide to move forward.

**Reply by Ougadougou75** [reply]
@jgVitc did you get anywhere with this? I am facing a similar challenge (adding redaction annotations) so if you have some previous experience that you are willing to share, it would be great!

**Reply by APOSWE** [reply]
hello!
is there any new state for this annotation topic? @jgVitc @Ougadougou75 ?
would be great if such feature is implemented!
